presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java

```diff
@@ -157,11 +160,21 @@ public void checkCanSetUser(Principal principal, String userName)
     }
 
     @Override
+    public void checkCanAccessCatalog(Identity identity, String catalogName)
+    {
+        if (filterCatalogs(identity, ImmutableSet.of(catalogName)).isEmpty()) {
+            denyCatalogAccess(catalogName);
+        }
+    }
+
+    @Override
     public void checkCanCreateSchema(TransactionId transactionId, Identity identity, CatalogSchemaName schemaName)
     {
         requireNonNull(identity, "identity is null");
         requireNonNull(schemaName, "schemaName is null");
 
+        authenticationCheck(() -> checkCanAccessCatalog(identity, schemaName.getCatalogName()));
```

Comment:
Good point. I looked into this. I am not sure what all such methods are, that need to have the catalog access control check, if we move it out of `AccessControlManager`. For example, even if I add a check in both, `getCatalogHandle` and `getTableHandle`, there are still other statements that would be missed since they call none of these two methods (e.g. methods in `MetadataListing.java`).